src/controllers/LoginController.ts

```diff
@@ -18,62 +22,97 @@ export default class LoginController {
-    const existingUser = await User.findOne({ where: { email } })
-    if (!existingUser) {
-      return res.status(401).json({ message: 'Invalid email or password' })
-    }
-    if (existingUser.verified !== true) {
-      return res
-        .status(401)
-        .json({ message: 'Check your email and verify your account' })
-    }
+    try {
+      const { email, password } = req.body
 
-    const isPasswordValid = await comparePassword(
-      password,
-      existingUser.password,
-    )
-    if (!isPasswordValid) {
-      return res.status(401).json({ message: 'Invalid email or password' })
-    }
+      const existingUser = await User.findOne({ where: { email } })
 
-    createUserProfile(existingUser.id)
+      if (!existingUser) {
+        return res.status(404).send({ message: 'User not found' })
+      }
 
-    const tokenPayload: {
-      id: number
-      userRole: string
-      email: string
-      otp?: string
-    } = {
-      id: existingUser.id,
-      userRole: existingUser.userRole,
-      email: existingUser.email,
-    }
+      const isPasswordValid = await comparePassword(
+        password,
+        existingUser.password,
+      )
+      if (!isPasswordValid) {
+        return res.status(401).json({ message: 'Invalid email or password' })
+      }
+
+      createUserProfile(existingUser.id)
+
+      if (existingUser.dataValues.verified === false) {
+        // Handle pending verification
+        const pendingVerification = await Token.findOne({
+          where: { userId: existingUser.id },
+        })
 
-    if (existingUser.userRole === 'seller') {
-      const otp = `${Math.floor(1000 + Math.random() * 9000)}`
-      const html = otpEmailTemplate(existingUser.username, otp)
-      const otptoken = generateToken({
+        // Generate and send new verification email
+        const newVerificationToken = generateToken({
+          id: existingUser.id,
+          email: existingUser.email,
+        })
+        const baseUrl = `${process.env.NODEMAILER_BASE_URL}users/verify/${newVerificationToken}`
+        const html = verificationsEmailTemplate(existingUser.username, baseUrl)
+
+        if (!pendingVerification) {
+          await Token.create({
+            userId: existingUser.id,
+            token: newVerificationToken,
+          })
+        } else {
+          await pendingVerification.update({ token: newVerificationToken })
+        }
+
+        try {
+          await sendMail(email, 'verify Your Account', html)
+        } catch (error) {
+          return res.status(500).json({ message: 'Internal server error' })
+        }
+
+        return res
+          .status(200)
+          .send({ message: 'A verification email has been sent' })
+      }
+
+      const tokenPayload: {
+        id: number
+        userRole: string
+        email: string
+        otp?: string
+      } = {
         id: existingUser.id,
+        userRole: existingUser.userRole,
         email: existingUser.email,
```

Comment:
```suggestion
      const tokenPayload = {
        id: existingUser.id,
        userRole: existingUser.userRole,
        email: existingUser.email,
```
I guess the type  will be infered automatically